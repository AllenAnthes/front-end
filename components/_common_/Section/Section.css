.Section {
  align-items: center;
  display: flex;
  flex-direction: column;
  min-height: 250px;
  padding: 3rem 0;
  width: 100%;
}

.content {
  align-items: flex-start;
  display: flex;
  flex-flow: column;
  justify-content: space-between;
  width: 85%;
}

@media screen and (max-width: 768px) {
  .content {
    width: 90%;
  }
}

/* Dark Themes */
.secondary {
  background-color: var(--secondary);
  color: var(--white);
}

/* Light Themes */
.gray,
.white {
  color: var(--secondary);
}

.gray {
  background-color: var(--gray);
}

.white {
  background-color: var(--white);
}
