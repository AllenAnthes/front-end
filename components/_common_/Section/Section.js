import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Heading from 'components/_common_/Heading/Heading';
import styles from './Section.css';

Section.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  hasHeadingLines: PropTypes.bool,
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]), // reference for scroll anchors
  style: PropTypes.object,
  theme: PropTypes.oneOf(['gray', 'secondary', 'white']),
  title: PropTypes.string,
  contentClassName: PropTypes.string,
};

Section.defaultProps = {
  className: undefined,
  hasHeadingLines: true,
  id: '',
  style: {},
  theme: 'gray',
  title: '',
  contentClassName: undefined,
};

function Section({
  children,
  className,
  hasHeadingLines,
  id,
  theme,
  style,
  title,
  contentClassName,
}) {
  // heading theme should contrast from section's theme.
  let headingTheme;
  switch (theme) {
    // Light background colors have dark heading color
    case 'white':
    case 'gray':
      headingTheme = 'secondary';
      break;
    // Dark background colors have light heading color
    case 'secondary':
      headingTheme = 'white';
      break;
    default:
      break;
  }

  return (
    <section className={classNames(className, styles.Section, styles[theme])} style={style}>
      {title && (
        <Heading id={id} hasHeadingLines={hasHeadingLines} theme={headingTheme}>
          {title}
        </Heading>
      )}
      <div className={`${styles.content} ${contentClassName}`}>{children}</div>
    </section>
  );
}

export default Section;
