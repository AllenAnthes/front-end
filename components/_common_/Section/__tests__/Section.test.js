/* eslint-env jest */
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import Section from '../Section';

describe('Section', () => {
  it('should render with required props', () => {
    createSnapshotTest(<Section>Test Children</Section>);
  });

  it('should render with many props assigned', () => {
    createSnapshotTest(
      <Section
        className="testing-123"
        hasHeadingLines={false}
        id="test-1"
        style={{ color: 'black' }}
        theme="gray"
        title="Test"
        contentClassName="TestClassName"
      >
        Test Children 2
      </Section>,
    );
  });

  it('should render common use case', () => {
    createSnapshotTest(
      <Section theme="white" title="Test 2">
        Test Children 3
      </Section>,
    );
  });
});
