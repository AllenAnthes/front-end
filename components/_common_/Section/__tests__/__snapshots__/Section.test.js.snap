// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should render common use case 1`] = `
<section
  className="Section white"
  style={Object {}}
>
  <h2
    className="Heading secondary headingLines"
    id=""
  >
    Test 2
  </h2>
  <div
    className="content undefined"
  >
    Test Children 3
  </div>
</section>
`;

exports[`Section should render with many props assigned 1`] = `
<section
  className="testing-123 Section gray"
  style={
    Object {
      "color": "black",
    }
  }
>
  <h2
    className="Heading secondary"
    id="test-1"
  >
    Test
  </h2>
  <div
    className="content TestClassName"
  >
    Test Children 2
  </div>
</section>
`;

exports[`Section should render with required props 1`] = `
<section
  className="Section gray"
  style={Object {}}
>
  
  <div
    className="content undefined"
  >
    Test Children
  </div>
</section>
`;
